File: python/BubbleSort/bubble_sort_for.py
# Bubble Sort For Loop
# @author: Shubham Heda
# Email: hedashubham5@gmail.com

def main():
    print("**NOTE**: Elements should be less then 25 as alogrithm work best on that only")
    print("Enter Bubble Sort elements sepreated by (',') Comma: ")
    input_string = input()
    array = [int(each) for each in input_string.split(",")]
    array_len = len(array)
    if array_len > 1:
        for _ in range(array_len):
            swapped = False
            for col in range(array_len - 1):
                if array[col] > array[col + 1]:
                    array[col], array[col + 1] = array[col + 1], array[col]
                    swapped = True
                    print("New Array: ", array)
            if not swapped:
                break
    else:
        print("Array Contains Only One Value: ", array)
    print("===========================")
    print("Sorted Array: ", array)

if __name__ == '__main__':
    main()

File: rust/BubbleSort/bubble_sort_for.rs
/*
# Bubble Sort For Loop
# @author: Shubham Heda
# Email: hedashubham5@gmail.com
 */

use std::io::{self};

fn main() {
    println!("Bubble Sort Algorithm");
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer).unwrap();
    let mut array:Vec<i32> = buffer.trim().split(',').map(|s| s.parse::<i32>().unwrap()).collect();
    let array_len = array.len();
    println!("Array length {:?}", array_len);
    println!("Unsorted Array {:?}", array);
    println!("===========================");
    if array_len > 25 {
        panic!("Array Size should be less than 25 as larger array increases time complexity of execution");
    }
    if array_len >  1 {
        for _ in 0..array_len {
            let mut swapped = false;
            for col in 0..(array_len - 1) {
                if array[col] > array[col+1] {
                    let val = array[col + 1];
                    array[col + 1] = array[col];
                    array[col] = val;
                    swapped = true;
                    println!("New Array {:?}", array);
                }
            }
            if !swapped {
                break;
            }
        }
    } else {
        println!("Array Contains only one Value: {:?}", array);
    }
    println!("===========================");
    println!("Sorted Array: {:?}", array);

}